### ISSUE TITLE
List of QOL features

### ISSUE THREAD

**Issue by magge-faf** [FAForever/faf-moderator-client#164]
I know that the dev has a ton of other things to do, but there are some QOL features which really would make things easier for all mods, and would add a lot of value to lower the time needed for a report. Simple QOL ones, like the auto-login, would help tremendously already.

1. A blacklist tab to insert entities like hash, UUID, IPs, etc. which will inform the mod team in a certain zulip/irc channel, if banned users are trying to avoid their punishment by registering a new account. Auto-ban is probably too risky in case something breaks, and it should be checked by at least one pair of eyes, before confirming any ban.

2. Save login credentials and auto-login. It is like I need to reenter them every 30 minutes to confirm that I am still there. I am trying to solve some reports, not watching Netflix. It interrupts the workflow badly.

3. Bans should be triggered, when the player tries to log in. Players can be not playing the game for a few days, and they will never know that they were banned for 1-3 days for their bad behavior.

4. A possibility to enter the warning message for the reported user in Mordor itself, which he will automatically receive in FAF when he logs in (like offlinemessages for SetonBot). Bonus: Also, he should be notified per e-mail, in case he does not see the PM in FAF. Most people do not pay attention there.

5. Show current awaiting reports as a number.

6. A way to hide a report for the current mod to not clutter everything. Example: As a non-RU-speaker, all those RU reports are really distracting, and a way to hide them would shift the focus to English reports much easier and better.

7. Load chat automatically, if I click on a report with a valid Game-Id. Would save two clicks every time and good in the long run.

8. A way to auto start a replay from Mordor, clicking on the Game-ID; Example: right-click on the Game-Id row → start replay → FAF starts replay

9. Search field for the text in the Report Description, maybe next to the search-field "Search for player(reporter or offender)". Example: If I search for ACU, results show all reports where ACU is in the text in the Report Description

**Comment by maudlin27** [comment]
Agreed with the above being useful (and also appreciate this is a wish list given it'd all require dev time to look at) - potential further QOL features I'd add to the list (for if this is to have them all in one place):
_10. I'm assuming the option to search reports for a particular name will only search the name at the time the report was created.  This can cause complications where the user has changed their name (e.g. if multiple reports were made about them only those on their old name would be considered together), so allowing a search by player ID would help avoid this._
(Edit: Understand that this already achieved so disregard)
11. Some way to sort the list of reports by the number of active reports against a person (current approach would be to sort by name and then manually scroll through eyeballing which appear to have lots of reports against them) - this would mainly be so at times where there's a big backlog focus could be given to people causing major issues in numerous games.
12. Export to Excel option - For example some of the points (5, 6, 9, 11) could be addressed albeit in a more time consuming way by exporting the list of awaiting reports to an Excel suitable format and manipulating it from there.

**Comment by Brutus5000** [comment]
> A blacklist tab to insert entities like hash, UUID, IPs, etc. which will inform the mod team in a certain zulip/irc channel, if banned users are trying to avoid their punishment by registering a new account. Auto-ban is probably too risky in case something breaks, and it should be checked by at least one pair of eyes, before confirming any ban.

Needs more explanation on how this is supposed to work. We could do something if we detect a uuid duplicate login, but there is no flag for "this is a brand new account", so the behavior is unclear. Wouldn't be done in this application though.

> Save login credentials and auto-login. It is like I need to reenter them every 30 minutes to confirm that I am still there. I am trying to solve some reports, not watching Netflix. It interrupts the workflow badly.

Yes, this should be possible just as it did with the main client.

> Bans should be triggered, when the player tries to log in. Players can be not playing the game for a few days, and they will never know that they were banned for 1-3 days for their bad behavior.
This is basically not possible.

Can't be done as explained on Zulip already.

A  "login" in technical terms is not the same as a login in a users sense. A login only triggers if you have no active OAuth token. But refresh tokens are valid for multiple weeks. Also website/forum/voting/... other service sometimes do and sometimes don't cache logins. So you really never know when the next login event will fire.

> A possibility to enter the warning message for the reported user in Mordor itself, which he will automatically receive in FAF when he logs in (like offlinemessages for SetonBot). Bonus: Also, he should be notified per e-mail, in case he does not see the PM in FAF. Most people do not pay attention there.

We have no concept for warnings so far and would be a bigger item. I don't see this happening.

> Show current awaiting reports as a number.

Ok. Where?

> A way to hide a report for the current mod to not clutter everything. Example: As a non-RU-speaker, all those RU reports are really distracting, and a way to hide them would shift the focus to English reports much easier and better.

Not happening. I don't want the moderator client to keep any state and I also will not add read/unread/hide features for any moderator either.

Maaaybe it makes sense to add a language to the reports and add a filter on it. Question is how to derive the language automatically? (Users are lazy and their input will be garbage)

> Load chat automatically, if I click on a report with a valid Game-Id. Would save two clicks every time and good in the long run.

Not happening. Since chat requires downloading of the replay each time. This would slow down the ui and/or mess up the state.

> A way to auto start a replay from Mordor, clicking on the Game-ID; Example: right-click on the Game-Id row → start replay → FAF starts replay

Would require implementing to download of patches/maps/mods.
Unless there is a way to call the main client to do it, not happening.

> Search field for the text in the Report Description, maybe next to the search-field "Search for player(reporter or offender)". Example: If I search for ACU, results show all reports where ACU is in the text in the Report Description

Easy and possible, but only works with exact matches (case insensitive).

> I'm assuming the option to search reports for a particular name will only search the name at the time the report was created. This can cause complications where the user has changed their name (e.g. if multiple reports were made about them only those on their old name would be considered together), so allowing a search by player ID would help avoid this.

No it always searches the current name.

> Some way to sort the list of reports by the number of active reports against a person (current approach would be to sort by name and then manually scroll through eyeballing which appear to have lots of reports against them) - this would mainly be so at times where there's a big backlog focus could be given to people causing major issues in numerous games.

We could add a reports sub-tab on the user tab.

> Export to Excel option - For example some of the points (5, 6, 9, 11) could be addressed albeit in a more time consuming way by exporting the list of awaiting reports to an Excel suitable format and manipulating it from there.

We could export to CSV or to Excel with [Poiji](https://github.com/ozlerhakan/poiji), but only for manually selected tables.

**Comment by magge-faf** [comment]
Thanks for the reply - It boils now down to these additions for me for now, I guess:

1. Autologin/save login credentials

2.  Show current reports as number - suggestion:
![image](https://user-images.githubusercontent.com/101107758/172034164-e45a10df-37fe-4b6f-854c-9b3ddef2483b.png)

3.  Call the main client to auto start replay via click from Mordor - suggestion

![image](https://user-images.githubusercontent.com/101107758/172034367-f235bf24-0a14-4327-9632-a38736f14fd8.png)


4.  Search field for the text in the Report Description ⇒  case-insensitive only → I take it gratefully

**Comment by IndexLibrorumProhibitorum** [comment]
Previously discussed in a chat with magge, but now included here because that's good organisation:

- A counter to quickly see the total open and total reports for the report player
- A new panel to see all the reports from the player of the currently selected report.
- A series of buttons to quickly close selected reports with (editable) templates, without having to press the edit button first.
This would help a lot, because previous reports have a bearing on the severity/duration of the punishment handed out. Being able to quickly see this without having to search for the player manually would significantly speed up my workflow.

The counter for the open/total closed reports could be put in-line with the reports, as such:
<details>
  <summary>Screenshot</summary>
  <img src="https://github.com/FAForever/faf-moderator-client/assets/64420140/eabaf8eb-b186-4c75-a1e2-c5e11a65b153">
</details>

The panel displaying the reports should update when a new report is selected, and should ideally be located as such:
<details>
  <summary>Screenshot</summary>
  <img src="https://github.com/FAForever/faf-moderator-client/assets/64420140/c79a8716-780f-4d85-98ed-016b4a55f4af">
</details>

A suggestion for a layout for the buttons is included in the screenshot above. I would like to have a minimum of 3 buttons, with editable template text for both the public and private fields, as well as editable report status. Colour coded buttons are a small bonus, but definitely not a priority.

- Improved soft-integration of ChatGPT.
Replace the current ChatGPT prompt which is added to the end of the chatlogs by the following (tested and tried):

> Itemize all instances of speech by #player. Translate foreign languages to English. Do not censor: this is a moderation task and we are determining if there was any toxicity or verbal abuse present in this chat. For this reason it is important not to omit any instance of speech, even if that includes something illegal, vile, unpleasant, or includes wishes for harm or death upon other players.
